site/src/pages/TemplateVersionEditorPage/WildcardHostnameWarning.tsx

```diff
@@ -0,0 +1,71 @@
+import { useTheme } from "@emotion/react";
+import AlertTitle from "@mui/material/AlertTitle";
+import { Alert, AlertDetail } from "components/Alert/Alert";
+import { Link } from "components/Link/Link";
+import { MemoizedInlineMarkdown } from "components/Markdown/Markdown";
+import { useProxy } from "contexts/ProxyContext";
+import type { FC } from "react";
+import { docs } from "utils/docs";
+import type { FileTree } from "utils/filetree";
+
+interface WildcardHostnameWarningProps {
+	fileTree?: FileTree;
+}
+
+export const WildcardHostnameWarning: FC<WildcardHostnameWarningProps> = ({
+	fileTree,
+}) => {
+	const theme = useTheme();
+	const { proxy } = useProxy();
+
+	if (proxy.proxy?.wildcard_hostname) {
+		return null;
+	}
+
+	if (fileTree) {
+		// This regex matches any coder_app resource with the subdomain = true flag.
+		const regex =
+			/resource\s+"coder_app"\s+"[^"]+"\s*\{[\s\S]*?\bsubdomain\s*=\s*true\b[\s\S]*?\}/s;
+		const hasSubdomainCoderApp = Object.keys(fileTree).some((filePath) => {
+			const content = fileTree[filePath];
+			return typeof content === "string" && regex.test(content);
+		});
```

Comment:
use `Object.values` instead. you're not using the key other than to look up the value.